beacon-chain/core/blocks/deposit.go

```diff
@@ -190,16 +193,17 @@ func ProcessDeposit(beaconState state.BeaconState, deposit *ethpb.Deposit, verif
 			WithdrawableEpoch:          params.BeaconConfig().FarFutureEpoch,
 			EffectiveBalance:           effectiveBalance,
 		}); err != nil {
-			return nil, err
+			return nil, newValidator, err
 		}
+		newValidator = true
```

Comment:
Move this after `AppendBalance` ?

Not sure if we ever to return `nil, true, errors.New("appending balance fails")`

Reply:
The state was mutated by inserting a new validator. That is what this bool is meant to represent.

Reply:
true

Reply:
Some offline discussion is that the append validator to state should also add the balance as well in the same method. That would fix any ambiguity of where the boolean assignment should be.

Reply:
I would prefer to keep it separate. All operations should be only relevant to their particular fields in the state(validator,balances,etc). Batching them together complicates this, it ensures that when reading the spec and comparing it to our implementation, it is mostly identical.

Reply:
I'll open another draft PR and we can discuss this there. I think it can be relatively low impact, but would love feedback on whether or not we need to be identical to spec in implementation here.